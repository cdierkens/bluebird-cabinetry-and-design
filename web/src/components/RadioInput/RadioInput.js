import PropTypes from "prop-types";
import React from "react";

const RadioInput = ({ label, value, group, checked, onChange }) => (
  <div>
    <label className="flex items-center">
      <span className="relative h-4 w-4">
        <input
          className="absolute z-0 w-full h-full focus:outline-none focus:shadow-outline shadow-sm"
          type="radio"
          id={value}
          name={group}
          value={value}
          checked={checked}
          onChange={onChange}
        />
        <span
          className={`absolute z-10 rounded-full inline-block w-full h-full ${
            checked ? "bg-blue-dark" : "bg-white"
          }`}
        ></span>
      </span>
      <span className="font-sans text-sm ml-2">{label}</span>
    </label>
  </div>
);

RadioInput.propTypes = {
  checked: PropTypes.bool,
  label: PropTypes.string.isRequired,
  group: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  value: PropTypes.string.isRequired,
};

RadioInput.defaultProps = {
  checked: false,
  onChange: () => {},
};

export default RadioInput;
