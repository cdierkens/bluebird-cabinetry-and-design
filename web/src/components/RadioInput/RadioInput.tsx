import { useField } from "formik";
import React from "react";

interface RadioInputProps {
  label: string;
  name: string;
  value: string;
}

const RadioInput: React.FC<RadioInputProps> = ({ label, ...props }) => {
  const [field] = useField({
    ...props,
    type: "radio",
  });

  return (
    <div>
      <label className="flex items-center cursor-pointer">
        <span className="relative h-4 w-4">
          <input
            className="absolute z-0 w-full h-full focus:outline-none focus:ring shadow-md"
            type="radio"
            {...field}
            {...props}
          />
          <span
            className={`absolute z-10 rounded-full inline-block w-full h-full ${
              field.checked ? "bg-blue-dark" : "bg-white"
            }`}
          ></span>
        </span>
        <span className="font-sans text-sm ml-2">{label}</span>
      </label>
    </div>
  );
};

export default RadioInput;
