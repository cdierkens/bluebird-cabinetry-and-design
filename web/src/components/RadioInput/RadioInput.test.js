import { render } from "@testing-library/react";
import React from "react";
import RadioInput from "./RadioInput";

it("matches snapshot", () => {
  const { asFragment } = render(<RadioInput />);

  expect(asFragment()).toMatchSnapshot();
});
