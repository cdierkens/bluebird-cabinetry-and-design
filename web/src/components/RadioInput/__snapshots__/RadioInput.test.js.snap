// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="flex items-center cursor-pointer"
    >
      <span
        class="relative h-4 w-4"
      >
        <input
          class="absolute z-0 w-full h-full focus:outline-none focus:shadow-outline shadow-md"
          name="mock-group"
          type="radio"
          value="mock-value"
        />
        <span
          class="absolute z-10 rounded-full inline-block w-full h-full bg-white"
        />
      </span>
      <span
        class="font-sans text-sm ml-2"
      >
        mock-label
      </span>
    </label>
  </div>
</DocumentFragment>
`;
